hooks/useClickChat.tsx

```diff
@@ -1,16 +1,20 @@
 import { useRouter } from "next/navigation";
 import useIsMobile from "./useIsMobile";
+import { useArtistProvider } from "@/providers/ArtistProvider";
 
 const useClickChat = () => {
   const { push } = useRouter();
   const isMobile = useIsMobile();
+  const { setIsArtistBeingCreated } = useArtistProvider();
+
 
   const handleClick = (conversation: any, toggleModal: () => void) => {
     if (isMobile) toggleModal();
     if (conversation?.agentId) {
       push(`/funnels/${conversation?.type}/${conversation.agentId}`);
       return;
     }
+    setIsArtistBeingCreated(false);
```

Comment:
I'm not liking that you need to use `setIsArtistBeingCreated` across multiple different files.
- I'd prefer the `isArtistBeingCreated` state be `self-contained` without needing to manage the state in multiple, unrelated, components.

Would the files being changed be `reduced` if you
- make a new hook `useCreateArtist`
- add a useEffect to setIsArtistBeingCreated to `false` on route change
- import the `useCreateArtist` hook in `useArtists` so it can be accessed globally in the provider